2024.4 (#32142)
home-assistant/home-assistant.github.io: source/_posts/2024-04-03-release-20244.markdown

### Filter panel

Previously, our automations page allowed you to filter by {% term areas %},
{% term entities %}, or {% term devices %}, but only one at a time, and the
filters were tucked away in a dropdown menu, which meant you needed to select
a dropdown menu inside a dropdown menu to filter by area.

A new filter panel fixes these issues by providing a new dual pane user
interface that can let you check multiple filters across multiple filter
types on the left that stay put and view the filtered results on the right.

__TODO: Insert screenshot showing the filters active on the device automation
page.__

We also introduced multiple new filter types for each table. For example,
{% term automations %} can now be filtered not only by {% term areas %},
{% term entities %} , or {% term devices %}, but also by used
{% term blueprints %}, labels, and categories (more on these later!).

We didn’t stop at just that page! The devices and entities pages now also have
a way to filter not only by state but also by integrations and areas. Almost all
tables now have useful new filters to help you find what you are looking for.

### Item grouping

> Browsing a list of hundreds of {% term automations %}  or thousands of 

Reviewer: ```suggestion
Browsing a list of hundreds of {% term automations %} or thousands of 
```

---
layout: post
title: "2024.4: Organize all the things!"

> description: "Time for spring cleaning 🧹 Introducing new ways to organize your automation (and more) with floors, labels, and categories. Easily add Matter devices from other controllers, and two brand new map and webpage dashboards."

Reviewer 2: ```suggestion
description: "Time for spring cleaning 🧹 Introducing new ways to organize your automations (and more) with floors, labels, and categories. Easily add Matter devices from other controllers, and two brand new map and webpage dashboards."
```

{% term entities %} , or {% term devices %}, but also by used
{% term blueprints %}, labels, and categories (more on these later!).

We didn’t stop at just that page! The devices and entities pages now also have
a way to filter not only by state but also by integrations and areas. Almost all
tables now have useful new filters to help you find what you are looking for.

### Item grouping

Browsing a list of hundreds of {% term automations %}  or thousands of 
{% term entities %} can be overwhelming. While filters can help narrow the list,
visually grouping the items into smaller chunks makes browsing easier.

With the item grouping feature, we introduced a new dropdown menu to pick
how you’d like the items on the table to be grouped. On the automations page,
you can group automations by their {% term state %} - whether enabled
or not - and custom categories (more on that later!). On the {% term devices %}
page, you can even group them by manufacturers, {% term areas %},
or {% term integrations %}, which is really cool because it makes the page much
easier to browse. It gives you a new perspective on organizing your vast
network of devices.

> <img class="no-shadow" src='/images/blog/2024-04/group-items-by-manufacturer.png' alt='Screenshots showing the devices pages, grouped by manufacturer.'>

Reviewer 2: ```suggestion
<img class="no-shadow" src='/images/blog/2024-04/group-items-by-manufacturer.png' alt='Screenshots showing the devices page, grouped by manufacturer.'>
```

### Selection mode and batch actions

We removed the clutter of displaying a column of checkboxes by introducing
a new selection mode. When it is toggled on, checkboxes will appear, and you
can apply actions to all your selections with one click on the toolbar.

<p class='img'>
    <img src="/images/blog/2024-04/enable-selection-mode.png" alt="Screenshots point out the enable selection mode button in the toolbar of the tables in Home Assistant."/>
    This button enables selection mode in the new tables.
</p>

It allows you to, for example, enable or disable multiple selected
{% term entities %} on the entities page at once. This new paradigm will allow
us to introduce more batch management actions in the future.

<img class="no-shadow" src='/images/blog/2024-04/bulk-actions.png' alt='Screenshots showing selection mode active in a table, with some items selected, highlighting the bulk actions you can perform on those selected items.'>

## Three new ways to organize

Upon reading through and ideating on the feature request thread on
[“Automation Grouping”](https://community.home-assistant.io/t/grouping-automations-on-frontend-for-organization/102771),

> we realize that organization can be done in multiple ways for different purposes.

Reviewer 2: ```suggestion
we realized that organization can be done in multiple ways for different purposes.
```

> we realize that organization can be done in multiple ways for different purposes.

Reviewer 3: ```suggestion
we realized that organization can be done in multiple ways for different purposes.
```

{% term Floors %} are a new way to organize your areas per level of floors in
your home. The better Home Assistant knows your home and how it is spatially
structured, the better it can help you. Especially when it comes to future
features like generating dashboards, voice commands, and maybe even
AI-related features.

<img class="no-shadow" src='/images/blog/2024-04/floors.png' alt='Screenshots showing areas settings page, which now also shows the areas grouped by floor.'>

You can use floors in your {% term automations %} and {% term scripts %} as a
target for your actions or control them with your voice. For example, you can
turn off all the lights on the downstairs floor when you go to bed or ask
[Assist](/voice_control/) to turn on the lights upstairs.

For our power users, you can even use [floors in your templates](/docs/configuration/templating/#floors)
to create the most advanced and creative automations.

{% my areas badge %}

### Labels: Tag everything any way you want

{% term Floors %} and {% term areas %} are great for representing your actual
home, but as everyone has their own ways of organizing their home, everyone

> has their own ways of organizing the Home Assistant, too!

Reviewer 2: ```suggestion
has their own ways of organizing their Home Assistant, too!
```

and it can be used to group things visually specific to a certain table
in Home Assistant.

This is great for those who have a particular way of displaying their
{% term entities %} by separating them into multiple sections on a specific page.
For example, on the {% term automations %} page, you can create categories only
used for visually grouping automations but nowhere else, like "Notifications"
or “NFC tags”. You can then view your automations grouped or filtered by
those categories.

__TODO: Show screenshot with automations grouped by categories__

As these categories are unique for each dashboard, you can have different
sets of categories depending on the place you are organizing. This means
you can have different categories on the automations page than the
{% term scene %}, {% term scripts %}, or helpers settings page.

### Time to clean up your Home Assistant!

In summary, there are a total of 4 taxonomies for organizing your entities.
It sounds like a lot, but they all serve a distinct purpose. We are extending
and clarifying {% term areas %} with {% term floors %}, and we are introducing
{% term labels %} and {% term categories %} to make your {% term automations %}

> and many other tables to look more organized and to be more manageable.

Reviewer 2: ```suggestion
and many other tables look more organized and be more manageable.
```

  features.

Last of all, to help us to help you to perfect the design of these new features,
we would love to hear your feedback and see how you use them! Please feel free
to share them in the comments below and discuss them in our community.

## Map dashboard

Introducing a new dashboard this release: The **map**! 🗺️

You might think, "Hey, we already have a map dashboard in Home Assistant!"
and you are right! However, the map was an integration that was always there
and didn't provide the flexibility and features that were wanted.

With the map dashboard's introduction, you could add multiple map dashboards
with different entities and configuration settings. For example, create a
dashboard to show the location of your family members, another to track your
car, and another to show the location of your pets. 🐶

<img class="no-shadow" src='/images/blog/2024-04/dashboard-map.png' alt='Screenshots showing the new map dashboard you can add when you add a new dashboard to your Home Assistant.'>

On upgrade, Home Assistant will automatically migrate your existing map
integration to a map dashboard.

You don't like having the map in your sidebar? Well, that is no longer a

> problem. Since it is now a dashboard it means you can remove in the dashboard

Reviewer 2: ```suggestion
problem. Since it is now a dashboard it means you can remove it in the dashboard
```

settings.

## Webpage dashboard

Another new dashboard for this release is the **webpage**! 🌐

The webpage dashboard allows you to add and embed a webpage to your dashboard.
This could be a web page from the internet or a local web page from a local
machine or device like your router or NAS.

> <img class="no-shadow" src='/images/blog/2024-04/dashboard-webpage.png' alt='Screenshots adding a new webpage dashboard to Home Assistant, embeding the Home Assistant website.'>

Reviewer 2: ```suggestion
<img class="no-shadow" src='/images/blog/2024-04/dashboard-webpage.png' alt='Screenshots showing addition of a new webpage dashboard to Home Assistant, embedding the Home Assistant website.'>
```

This dashboard replaces the old iFrame panel (`iframe_panel`). If you have
existing panels configured in your YAML configuration, Home Assistant will
automatically migrate them to the new webpage dashboard on upgrade.

> <img class="no-shadow" src='/images/blog/2024-04/embeded-home-assistant-website.png' alt='Screenshot showing the Home Assistant website embeded into the Home Assistant frontend using a webpage dashboard.'>

Reviewer 2: ```suggestion
<img class="no-shadow" src='/images/blog/2024-04/embeded-home-assistant-website.png' alt='Screenshot showing the Home Assistant website embedded into the Home Assistant frontend using a webpage dashboard.'>
```

Note that not every webpage can be embedded due to security restrictions that
some sites or pages have in place, enforced by your browser, that prevent
embedding them into a Home Assistant dashboard.

## Define the columns in the section view

In the last release, we [introduced the section view](/blog/2024/03/06/release-20243/#a-new-experimental-sections-view)
with drag and drop support. We are absolutely thrilled to see how many of you
loved it! 🥰

Many thanks for all the feedback and suggestions! We are prioritizing them and
are working on making the section view even better in the upcoming releases.

In this release, based on your feedback, we have added the ability for you to

> define the maximum number of columns in the section view uses. This allows you

Reviewer 2: ```suggestion
define the maximum number of columns that the section view uses. This allows you
```

making this a more accurate representation of the actual startup time.

Want to learn about the individual startup times of your integrations? You can
find them in the {% my repairs title="**Settings** -> **System** -> **Repairs**" %}
page by selecting the tree-dotted menu in the top right and selecting the
**Integration startup time** item.

{% my repairs badge %}

[@bdraco]: https://github.com/bdraco
[@balloob]: https://github.com/balloob

## Other noteworthy changes

There are many more improvements in this release; here are some of the other
noteworthy changes this release:

- [@karwosts] has improved your Home Assistant {% my profile title="user profile" %}
  page, making navigating and finding the settings you are looking for easier.
  Nice work!
- If you use a [sentence trigger] in your automation to trigger it using
  [Assist]. The trigger will now include the `device_id` in its trigger data
  you can automate with. Thanks, [@synesthesiam]!
- The [HomeWizard Energy] integration now supports the newest generation of
  HomeWizard Energy Sockets! Thanks, [@DCSBL]!
- We can now ship new releases of Home Assistant in waaay less time.

>   Building and shipping Home Assistant for you, when down from 1.5-2 hours to

Reviewer 2: ```suggestion
  Building and shipping Home Assistant for you, went down from 1.5-2 hours to
```

[#112449]: https://github.com/home-assistant/core/pull/112449

{% enddetails %}

{% details "Met Office" %}

The Met Office integration previously created two entities for each configured
location: one provided daily weather forecasts and the other hourly forecasts.

The Met Office integration now only creates a single entity that provides both
daily and hourly weather forecasts.

([@gjohansson-ST] - [#112452]) ([documentation](/integrations/metoffice))

[@gjohansson-ST]: https://github.com/gjohansson-ST
[#112452]: https://github.com/home-assistant/core/pull/112452

{% enddetails %}

{% details "Modbus" %}

The Modbus integration does not allow a configuration with the device only
(no entities); this will create an issue, and the entry will not be loaded.

For the special cases (like some custom_components) where it is wanted,
please add a dummy entity.

([@janiversen] - [#113516]) ([documentation](/integrations/modbus))

[@janiversen]: https://github.com/janiversen
[#113516]: https://github.com/home-assistant/core/pull/113516

{% enddetails %}

{% details "MQTT" %}

Support for `brightness` is now assumed for `mqtt` lights with the `json` schema

> unless only coelo modes `rgb`, `rgbw`, or `rgbww` are supported.

Reviewer 2: ```suggestion
unless only color modes `rgb`, `rgbw`, or `rgbww` are supported.
```

For light which only support  color modes `rgb`, `rgbw`, or `rgbww`, 
the `brightness` option flag can still be set to `false` if brightness is not
supported, in this case brightness support is emulated in Home Assistant by

> scaling the RGBx-values.

Reviewer 2: ```suggestion
For lights which only support  color modes `rgb`, `rgbw`, or `rgbww`, 
the `brightness` option flag can still be set to `false` if brightness is not
supported. In this case brightness support is emulated in Home Assistant by
scaling the RGBx-values.
```

For all other color modes except `ColorMode.ONOFF`, it is assumed brightness is
supported, and the `brightness` attribute will be included in the MQTT payload
when the `brightness` attribute is supplied to the `light.turn_on` service.

([@jbouwh] - [#111676]) ([documentation](/integrations/mqtt))

[@jbouwh]: https://github.com/jbouwh
[#111676]: https://github.com/home-assistant/core/pull/111676

{% enddetails %}

{% details "myUplink" %}

If you have a Nibe SMO 20 in your account, you will notice that some 550
entities are not provided by the integration.

> You can safely remove these entities as the integration does not use them.

Reviewer 2: ```suggestion
entities are no longer provided by the integration.

You can safely remove these entities as the integration does not use them.
```

- [Climate entity auxiliary heater is deprecated](https://developers.home-assistant.io/blog/2024/03/10/climate-aux-heater-deprecated)
- [Config Entries can now provide a reconfigure step](https://developers.home-assistant.io/blog/2024/03/21/config-entry-reconfigure-step)
- [Deprecate hass.helpers usage](https://developers.home-assistant.io/blog/2024/03/30/deprecate-hass-helpers/)
- [Deprecating `async_run_job` and `async_add_job`](https://developers.home-assistant.io/blog/2024/03/13/deprecate_add_run_job)
- [Integrations import in the executor to avoid blocking the event loop](https://developers.home-assistant.io/blog/2024/03/09/import_executor_default)
- [New state timestamp State.last_reported](https://developers.home-assistant.io/blog/2024/03/20/state_reported_timestamp)
- [Raising exceptions with translations](https://developers.home-assistant.io/blog/2024/03/15/exception-translations)

[devblog]: https://developers.home-assistant.io/blog/

## Farewell to the following

The following integrations are also no longer available as of this release:

- **OpenCV** has been removed. The integration has been in a broken state

>   for over an year and currently has no path forward.

Reviewer 2: ```suggestion
  for over a year and currently has no path forward.
```

We also introduced multiple new filter types for each table. For example,
{% term automations %} can now be filtered not only by {% term areas %},
{% term entities %} , or {% term devices %}, but also by used

> {% term blueprints "blueprints," %} labels, and categories (more on these later!).

Reviewer 2: Is the double blueprints correct?

Author: Yes, it denotes the term to look up vs what to print on the page.

In this case, the `,` gets lost with an additional new-line on a desktop. Not sure why, never seen it before. This works around it.

Author: But it doesn't work so got to revert it :)
